src/SupportInfillPart.cpp

```diff
@@ -14,27 +14,34 @@ SupportInfillPart::SupportInfillPart(const PolygonsPart& outline, coord_t suppor
 , inset_count_to_generate(inset_count_to_generate)
 {
     insets.clear();
+    wall_toolpaths.clear();
     infill_area.clear();
     infill_area_per_combine_per_density.clear();
 }
 
 bool SupportInfillPart::generateInsetsAndInfillAreas()
 {
-    // generate insets, use the first inset as the wall line, and the second as the infill area
-    AreaSupport::generateOutlineInsets(insets, outline, inset_count_to_generate, support_line_width);
-    if (inset_count_to_generate > 0 && insets.empty())
+    if (inset_count_to_generate == 0)
     {
-        return false;
+        infill_area = outline;
+        infill_area.simplify();
     }
-
-    // calculate infill area if there are insets
-    if (inset_count_to_generate > 0)
+    else
     {
-        // if there are walls, we use the inner area as the infill area
-        infill_area = insets.back().offset(-support_line_width / 2);
-        // optimize polygons: remove unnecessary verts
+        constexpr coord_t smallest_segment = 50;
+        constexpr coord_t allowed_distance = 50;
+        constexpr coord_t epsilon_offset = 10;
+        constexpr float max_colinear_angle = 0.03;
+        constexpr bool remove_holes = false;
+        const double small_area_length = INT2MM(static_cast<double>(support_line_width) / 2);
+
+        prepared_outline = outline.offset(-epsilon_offset).offset(epsilon_offset);
+        prepared_outline.simplify(smallest_segment, allowed_distance);
+        prepared_outline.removeColinearEdges(max_colinear_angle);
+        prepared_outline.fixSelfIntersections();
+        prepared_outline.removeSmallAreas(small_area_length * small_area_length, remove_holes);
+        infill_area = outline.offset(-static_cast<coord_t>(small_area_length) - static_cast<coord_t>(support_line_width) * (inset_count_to_generate));
```

Comment:
There's a lot of magic here. Could you document what each of these steps are doing and why they are necessary? And why are those hard-coded parameters chosen?

Like, is it necessary for the Voronoi diagram to be generated or something? Maybe it would be good to have a re-usable function for that, or add it to the routine that actually generates the Voronoi.

Reply:
Agreed with the use of a reusable function, for now these parameters and steps are copied how @rburema handles the walls, they need to be fine-tuned at a later time. The current values are "dikke duim" values that seem to work. I could add `// todo: magic number` So they are easier to find when we will fine-tune those parameters.